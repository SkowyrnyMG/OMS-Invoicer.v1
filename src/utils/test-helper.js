import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
