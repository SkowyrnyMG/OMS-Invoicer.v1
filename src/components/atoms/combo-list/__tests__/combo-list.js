import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import ComboList from '../combo-list';

describe('ComboList', () => {
  test('should display in the document with correct styels', () => {
    snapShotTest(<ComboList />);
  });
});
