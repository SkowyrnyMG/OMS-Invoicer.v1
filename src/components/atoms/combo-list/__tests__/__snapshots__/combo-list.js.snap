// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboList should display in the document with correct styels 1`] = `
.c0 {
  position: absolute;
  max-height: 25rem;
  padding: 2rem;
  list-style: none;
  overflow-y: auto;
  background: #F8F8FF;
  box-shadow: 0 4px 5px rgba(0,0,0,0.1);
  opacity: 0;
  visibility: hidden;
  z-index: 1000;
}

<ul
  class="c0"
/>
`;
